x-pack/libbeat/management/manager.go

```diff
@@ -195,6 +192,11 @@ func (cm *ConfigManager) worker() {
 		}
 
 		if firstRun {
+			// Send metadata
+			if err := cm.client.UpdateMetadata(cm.config.Metadata); err != nil {
+				cm.logger.Errorf("Could not update metadata, error: %v", err)
+			}
```

Comment:
I am ok with the first version to not do a retry yet, I think we have to clarify which API command we want to retry and also we need to make sure we do it gracefully, we don't want 100 beats to hammer ES.